plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2932,17 +2936,31 @@ public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession s
                 .orElseThrow(() -> new IllegalStateException("Storage table missing in definition of materialized view " + materializedViewName));
 
         Table icebergTable = catalog.loadTable(session, storageTableName);
-        String dependsOnTables = Optional.ofNullable(icebergTable.currentSnapshot())
+        Snapshot currentSnapshot = icebergTable.currentSnapshot();
+        String dependsOnTables = Optional.ofNullable(currentSnapshot)
                 .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_TABLES, ""))
                 .orElse("");
+        String dependsOnTableFunctions = Optional.ofNullable(icebergTable.currentSnapshot())
+                .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_FUNCTIONS, ""))
+                .orElse("");
+
+        Optional<Instant> refreshTime = Optional.ofNullable(currentSnapshot)
+                .map(snapshot -> snapshot.summary().get(TRINO_QUERY_START_TIME))
+                .map(Instant::parse)
+                .or(() -> Optional.ofNullable(currentSnapshot)
+                                .map(snapshot -> Instant.ofEpochMilli(snapshot.timestampMillis())));
+
+        if (!dependsOnTableFunctions.isEmpty()) {
+            // we will never know if table function returns stale value or not
+            return new MaterializedViewFreshness(UNKNOWN, refreshTime);
+        }
+
         if (dependsOnTables.isEmpty()) {
             // Information missing. While it's "unknown" whether storage is stale, we return "stale".
-            // Normally dependsOnTables may be missing only when there was no refresh yet.
+            // Normally dependsOnTables may be missing only when there was no refresh yet, when the source is a PTF or when there is no source for example: CREATE M V foo AS VALUES 1
```

Comment:
no need to talk about PTF here, since they are handled above.
i think you can just revert changes to this line